src/Uno.UI/UI/Xaml/UIElementGeneratedProxy.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Uno.UI.Xaml;
+
+public static class UIElementGeneratedProxy
+{
+	private static readonly Dictionary<Type, RoutedEventFlag> _implementedRoutedEvents = new();
+
+	public static RoutedEventFlag RegisterImplementedRoutedEvents(Type type, RoutedEventFlag routedEventFlags)
+	{
+		lock (_implementedRoutedEvents)
+		{
+			_implementedRoutedEvents[type] = routedEventFlags;
+			return routedEventFlags;
+		}
+	}
+
+	internal static bool TryGetImplementedRoutedEvents(Type type, out RoutedEventFlag routedEventFlags) =>
+		_implementedRoutedEvents.TryGetValue(type, out routedEventFlags);
```

Comment:
Is using https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2?view=net-6.0 better?

Reply:
I think @jeromelaban preferred the locking here

Reply:
Let's keep the lock for now until we validate.